pkg/api/tag_test.go

```diff
@@ -69,128 +64,140 @@ func TestTags(t *testing.T) {
-		_ = jsonhttptest.ResponseDirectSendHeadersAndReceiveHeaders(t, apiClient, http.MethodPost, resource(validHash), bytes.NewReader(validContent), http.StatusOK, jsonhttp.StatusResponse{
+		sentHeaders := make(http.Header)
+		sentHeaders.Set(api.TagHeaderUid, strconv.FormatUint(uuid, 10))
+		_ = jsonhttptest.ResponseDirectSendHeadersAndReceiveHeaders(t, client, http.MethodPost, chunksResource(validHash), bytes.NewReader(validContent), http.StatusOK, jsonhttp.StatusResponse{
 			Message: http.StatusText(http.StatusOK),
 			Code:    http.StatusOK,
-		}, sentHheaders)
+		}, sentHeaders)
 
 		// see if the tagid is present and has valid values
-		ta = debugapi.TagResponse{}
-		jsonhttptest.ResponseUnmarshal(t, ts.Client, http.MethodGet, tagResourceUUid(uuid), nil, http.StatusOK, &ta)
+		tr = api.TagResponse{}
+		jsonhttptest.ResponseUnmarshal(t, client, http.MethodGet, getTagResource(uuid), nil, http.StatusOK, &tr)
 
-		if uuid != uint64(ta.Uid) {
-			t.Fatalf("Invalid uuid response")
+		if uuid != uint64(tr.Uid) {
+			t.Fatalf("Invalid id response")
 		}
-		if ta.Stored != 2 {
+		if tr.Stored != 2 {
 			t.Fatalf("same tag not used")
 		}
 	})
 
-	t.Run("get-tag-using-uuid", func(t *testing.T) {
-		rcvdHeaders := jsonhttptest.ResponseDirectSendHeadersAndReceiveHeaders(t, apiClient, http.MethodPost, resource(validHash), bytes.NewReader(validContent), http.StatusOK, jsonhttp.StatusResponse{
+	t.Run("get-tag-using-id", func(t *testing.T) {
+		rcvdHeaders := jsonhttptest.ResponseDirectSendHeadersAndReceiveHeaders(t, client, http.MethodPost, chunksResource(validHash), bytes.NewReader(validContent), http.StatusOK, jsonhttp.StatusResponse{
 			Message: http.StatusText(http.StatusOK),
 			Code:    http.StatusOK,
 		}, nil)
-		uuid := isTagFoundInResponse(t, rcvdHeaders, nil)
+		id := isTagFoundInResponse(t, rcvdHeaders, nil)
 
-		// Request the tag and see if the UUID is the same
-		ta := debugapi.TagResponse{}
-		jsonhttptest.ResponseUnmarshal(t, ts.Client, http.MethodGet, tagResourceUUid(uuid), nil, http.StatusOK, &ta)
-		if uuid != uint64(ta.Uid) {
+		// Request the tag and see if the ID is the same
+		tr := api.TagResponse{}
+		jsonhttptest.ResponseUnmarshal(t, client, http.MethodGet, getTagResource(id), nil, http.StatusOK, &tr)
+		if id != uint64(tr.Uid) {
 			t.Fatalf("Invalid uuid response")
 		}
 	})
 
 	t.Run("tag-counters", func(t *testing.T) {
-		rcvdHeaders := jsonhttptest.ResponseDirectSendHeadersAndReceiveHeaders(t, apiClient, http.MethodPost, resource(validHash), bytes.NewReader(validContent), http.StatusOK, jsonhttp.StatusResponse{
+		rcvdHeaders := jsonhttptest.ResponseDirectSendHeadersAndReceiveHeaders(t, client, http.MethodPost, chunksResource(validHash), bytes.NewReader(validContent), http.StatusOK, jsonhttp.StatusResponse{
 			Message: http.StatusText(http.StatusOK),
 			Code:    http.StatusOK,
 		}, nil)
-		uuid1 := isTagFoundInResponse(t, rcvdHeaders, nil)
+		id := isTagFoundInResponse(t, rcvdHeaders, nil)
```

Comment:
`id := uint32(...)` then you can remove the casts below, or just return `uint32` from `isTagFoundIn...`

Reply:
done, thanks